event-log/src/main/scala/io/renku/eventlog/init/EventPayloadTableCreator.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.eventlog.init
+
+import cats.effect.Bracket
+import ch.datascience.db.{DbClient, DbTransactor}
+import ch.datascience.graph.model.events.EventStatus.GenerationRecoverableFailure
+import io.chrisdavenport.log4cats.Logger
+import io.renku.eventlog.EventLogDB
+
+private trait EventPayloadTableCreator[Interpretation[_]] {
+  def run(): Interpretation[Unit]
+}
+
+private object EventPayloadTableCreator {
+  def apply[Interpretation[_]](
+      transactor: DbTransactor[Interpretation, EventLogDB],
+      logger:     Logger[Interpretation]
+  )(implicit ME:  Bracket[Interpretation, Throwable]): EventPayloadTableCreator[Interpretation] =
+    new EventPayloadTableCreatorImpl(transactor, logger)
+}
+
+private class EventPayloadTableCreatorImpl[Interpretation[_]](
+    transactor: DbTransactor[Interpretation, EventLogDB],
+    logger:     Logger[Interpretation]
+)(implicit ME:  Bracket[Interpretation, Throwable])
+    extends EventPayloadTableCreator[Interpretation]
+    with EventTableCheck[Interpretation] {
+
+  import cats.syntax.all._
+  import doobie.implicits._
+
+  private implicit val transact: DbTransactor[Interpretation, EventLogDB] = transactor
+
+  override def run(): Interpretation[Unit] =
+    whenEventTableExists(
```

Comment:
Shouldn't the new table actually be created only if the `event` table exists?

Reply:
Yes good catch!